helper/resource/error.go

```diff
@@ -0,0 +1,21 @@
+package resource
+
+type NotFoundError struct {
+	LastError    error
+	LastRequest  interface{}
+	LastResponse interface{}
+	Message      string
+	Retries      int
+}
+
+func (e *NotFoundError) Error() string {
+	if e.Message != "" {
+		return e.Message
+	}
+
+	return "couldn't find resource"
+}
+
+func NewNotFoundError(err string) *NotFoundError {
```

Comment:
Good convo here so far!

My primary design goal for the `RetryableError` / `NonRetryableError` helpers is to _force_ the user to decide whether or not the error they're returning is retryable.

Struct initialization lets a user omit the `Retryable bool` field without being yelled at, catching the zero value.

There's probably a way to accomplish the same goal and allow struct literal client code using interface implementations.

In this case, there's less of a _"client code must fill in these fields"_ aspect, so I think going with the struct literal seems reasonable.

Does that make sense? Happy to discuss further!